bangazonapi/views/product.py

```diff
@@ -293,3 +303,56 @@ def recommend(self, request, pk=None):
             return Response(None, status=status.HTTP_204_NO_CONTENT)
 
         return Response(None, status=status.HTTP_405_METHOD_NOT_ALLOWED)
+
+    @action(methods=['post', 'delete'], detail=True)
+    def like(self, request, pk=None):
+        """Recommend products to other users"""
+
+        if request.method == "POST":
+            try:
+                customer = Customer.objects.get(user=request.auth.user)
+                product = Product.objects.get(pk=pk)
+                all_customer_likes = CustomerProductLike.objects.filter(customer_id=customer.id)
+                try:
+                    like = all_customer_likes.get(product_id=pk)
+                except:
+                    like = False
+                if not like:
+                    CustomerProductLike.objects.create(
+                        customer = customer,
+                        product = product
+                    )
+                    return Response(None, status=status.HTTP_201_CREATED)
+                else:
+                    return Response({"message":"Product already liked"}, status=status.HTTP_204_NO_CONTENT)
+            except Exception as ex:
+                return Response({'message': ex.args[0]})
+        elif request.method == "DELETE":
+            try:
+                customer = Customer.objects.get(user=request.auth.user)
+                all_customer_likes = CustomerProductLike.objects.filter(customer_id=customer.id)
+                like = all_customer_likes.get(product_id=pk)
```

Comment:
The `.get` function can take more than one parameter if you have multiple fields you want to get by. So line 333 could be combined with 334.
If the customer mistakenly likes a product more than once though, the get will probably fail with a `MultipleObjectsReturned` error. To solve that take a look at the django `unique_together` constraint for models.

Reply:
I have combined the get into one line. I already have some code in place to make sure a user cannot like a product more than once. Would I still need the unique together constraint added?

Reply:
I see now. Sorry I missed that part